lib/galaxy/workflow/extract.py

```diff
@@ -345,7 +345,7 @@ def cleanup( prefix, inputs, values ):
                 # Scrub dynamic resource related parameters from workflows,
                 # they cause problems and the workflow probably should include
                 # their state in workflow encoding.
-                if input.is_job_resource_conditional:
+                if input.name == '__job_resource':
                     if input.name in values:
```

Comment:
I'm going to merge this - but I don't understand your problem with giving names to repeated computations. You are duplicating the check and the resulting ``if`` check is both more obtuse and implementation dependent. As a result, this change makes things less readable and more easy to introduce bugs into.